cdap-docs/examples-manual/source/examples/sport-results.rst

```diff
@@ -0,0 +1,212 @@
+     partitioning field. Each record represents the total number of games won and lost, and the total number
+     of points scored and conceded, by a team.
+
+We will use the ``UploadService`` to upload files into the ``results`` dataset, then compute the ``totals``
+aggregates using MapReduce, and we will explore both datasets using SQL.
+
+UploadService
+-------------
+
+This service has two handler methods: one to upload and another to download a partition of the ``results``
+dataset as a file. It declares its use of the dataset using a ``@UseDataSet`` annotation:
+
+.. literalinclude:: /../../../cdap-examples/SportResults/src/main/java/co/cask/cdap/examples/sportresults/UploadService.java
+    :language: java
+    :lines: 56-59
+
+Let's take a closer look at the upload method:
+
+   - It first creates a partition key and a relative file path from the league and season
+     received as path parameters in the request URL.
+   - It then uses the ``getLocation()`` of the embedded file set of the ``results`` dataset to obtain the location
+     for writing the file, and opens an output stream for that location to write the file contents.
+     ``Location`` is a file system abstraction from `Apache™ Twill® <http://twill.incubator.apache.org>`__;
+     you can read more about its interface in the Apache Twill
+     `Javadocs <http://twill.incubator.apache.org/apidocs/org/apache/twill/filesystem/Location.html>`__.
+   - Finally, it registers the written file as a new partition in the dataset.
+
+.. literalinclude:: /../../../cdap-examples/SportResults/src/main/java/co/cask/cdap/examples/sportresults/UploadService.java
+    :language: java
+    :lines: 85-115
+
+MapReduce over File Partitions
+==============================
+
+``ScoreCounter`` is a simple MapReduce that reads from the ``results`` PartitionedFileSet and writes to
+the ``totals`` PartitionedFileSet. The ``beforeSubmit()`` method prepares the MapReduce program for this:
+
+   - It reads the league that it is supposed to process from the runtime arguments.
+   - It constructs a partition filter for the input using the league as the only condition, and instantiates
+     the ``results`` dataset with arguments that contain this filter.
+   - It constructs an output partition key for the new partition, and instantiates the ``totals`` dataset
+     with arguments that specify this partition key.
+
+.. literalinclude:: /../../../cdap-examples/SportResults/src/main/java/co/cask/cdap/examples/sportresults/ScoreCounter.java
+    :language: java
+    :lines: 60-86
+
+After the MapReduce has completed, the ``onFinish()`` method finalizes the addition of the new partition.
+Note that this is only needed due to a current limitation in the ``PartitionedFileSet`` implementation: itself
+should add the partition in the output committer of its output format (this will be addressed in a future CDAP release):
```

Comment:
this is your wording. You wrote this as a correction for what I wrote for the Stream Conversion example. It means, the dataset itself should do it. We could say instead: 

Note that this is only needed due to a current limitation in the ``PartitionedFileSet`` implementation: the partition should be added by the output committer of the dataset's output format (this will be addressed in a future CDAP release):